File: src/app/service/room.service.ts
import { HttpClient, HttpContext, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class RoomService {

  apiImportExcel = "http://localhost:8080/importExcel";
  apiTestGet = "http://localhost:8080/testGet"
  apiGetTimeStart = "http://localhost:8080/getStartTime"
  apiSetTimeStart = "http://localhost:8080/setStartTime"
  apiGetAllArea = "http://localhost:8080/getAllArea"
  apiGetAllRoomByArea = "http://localhost:8080/getAllRoomByArea"
  apiGetTimeAndWeekByRoomAndDay = "http://localhost:8080/getTimeAndWeekByRoomAndDay"
  apiGetAllRoom = "http://localhost:8080/getAllRoom"
  apiFindByRoomAndDay = "http://localhost:8080/findByRoomAndDay"
  apiSaveData = "http://localhost:8080/saveData"
  apiSaveRoom = "http://localhost:8080/saveRoom"

  constructor(private httpClient : HttpClient) { }

  importExcel(area : String, room : String, capacity : String, time : String, day : String , week:String){
    const body  = {area,room,capacity,time,day,week}
    return this.httpClient.post(this.apiImportExcel,body)
  }

  saveRoom(room : String,area : String,capacity : String){
    const body = {room,area,capacity}
    return this.httpClient.post(this.apiSaveRoom,body)
  }

  saveData(rooms : any){
    return this.httpClient.post(this.apiSaveData,rooms);
  }

  getAllArea(){
    return this.httpClient.get(this.apiGetAllArea)
  }

  getAllRoomByArea(area : String){
    const body = {area};
    return this.httpClient.post(this.apiGetAllRoomByArea,body);
  }

  getAllRoom(){
    return this.httpClient.get(this.apiGetAllRoom);
  }

  getTimeAndWeekByRoomAndDay(room : String, day : String){
    const body = {room,day}
    return this.httpClient.post(this.apiGetTimeAndWeekByRoomAndDay,body)
  }

  findByRoomAndDay(room : String, day: String){
    const body = {room,day};
    return this.httpClient.post(this.apiFindByRoomAndDay,body);
  }

  setTimeStart(timeStart : string, weekStart : string){
    const body = {timeStart,weekStart}
    return this.httpClient.post<any>(this.apiSetTimeStart,body,{
      headers : new HttpHeaders({
        'Content-type' : 'application/json'
      })
    })
  }

  getTimeStart(){
    return this.httpClient.get(this.apiGetTimeStart);
  }
}

File: qlgiangduong/src/main/java/com/example/qlgiangduong/Service/RoomService.java
package com.example.qlgiangduong.Service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.example.qlgiangduong.Entity.Room;
import com.example.qlgiangduong.Entity.Time;
import com.example.qlgiangduong.Reponsitory.RoomReponsitory;
import com.example.qlgiangduong.Reponsitory.TimeReponsitory;

@Service
public class RoomService {
    @Autowired
    private RoomReponsitory roomReponsitory;
    @Autowired
    private TimeReponsitory timeReponsitory;

    public boolean ImportExcel(Room room){
        roomReponsitory.save(room);
        return true;
    }

    public void SaveData(List<Room> rooms){
        roomReponsitory.deleteByArea(rooms.get(0).getArea());
        for (Room room : rooms) {
            roomReponsitory.save(room);
        }
    }

    public List<Time> getStartTime(){
        return timeReponsitory.findAll();
    }

    public void setStartTime(Time time){
        timeReponsitory.deleteAll();
        timeReponsitory.save(time);
    }

    public List<String> getAllArea(){
        return roomReponsitory.getAllArea();
    }

    public List<String> getAllRoomByArea(String area){
        return roomReponsitory.getAllRoomByArea(area);
    }

    public List<String> getAllRoom(){
        return roomReponsitory.getAllRoom();
    }

    public List<Object> getTimeAndWeekByRoomAndDay(String room, String day){
        return roomReponsitory.getTimeAndWeekByRoomAndDay(room,day);
    }

    public List<Room> findByRoomAndDay(String room, String day){
        return roomReponsitory.findByRoomAndDay(room, day);
    }
}
